### DISCUSSION TITLE
t is not loaded with dynamic url

### DISCUSSION THREAD

**Discussion by yazer79** [sveltekit-i18n/lib#137] [Q&A] [answered]
When I try to use a route like "...en/blog" with translation, it works fine
```   locale: 'en'
      key: 'blog',
      routes: ['/blog'],
      loader: async () => (await import('./en/blog.json')).default
```
But if it was a dynamic url with slug like " ...en/blog/amazing-post", the translation is not loaded
if you remove the slug, request the url to get a 404, then put the slug back again, the translation works with the dynamic route
Using: sveltekti18n 2.v.2 and locale-router example

**Comment by jarda-svoboda** [comment]
You could use RegExp instead of a string in `routes` array to match your dynamic route.

**Reply by yazer79** [reply]
I tried using regex. but didn't work for me: 
```routes: ['^blog/.*/$']```
```routes: ['blog/.*/']```
Does it require a modification in the hooks?

**Reply by jarda-svoboda** [reply, marked as answer]
Nope, use RegExp instead of a string like this:

```
routes: [/^\/blog\/.*/]
```

It’s demonstrated in [this example](https://github.com/sveltekit-i18n/lib/tree/master/examples/loaders).

**Reply by yazer79** [reply]
you mean:
```
loaders [
{ //...,
routes: [/^\/blog\/.*\/$/]
}
]
``` ? does not work with me

**Reply by jarda-svoboda** [reply]
Oh, sure, comment updated…)

**Comment by yazer79** [comment]
this worked for me:
```routes: [RegExp('/blog/(.*)')]```

[a comment by an automated account is not shown]
